frame/examples/dev-mode/src/lib.rs

```diff
@@ -0,0 +1,126 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! <!-- markdown-link-check-disable -->
+//! # Dev Mode Example Pallet
+//!
+//! A simple example of a FRAME pallet demonstrating
+//! the ease of requirements for a pallet in dev mode.
+//!
+//! Run `cargo doc --package pallet-dev-mode --open` to view this pallet's documentation.
+//!
+//! **Dev mode is not meant to be used in production.**
+//!
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::dispatch::DispatchResult;
+use frame_system::ensure_signed;
+
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+#[cfg(test)]
+mod tests;
+
+/// A type alias for the balance type from this pallet's point of view.
+type BalanceOf<T> = <T as pallet_balances::Config>::Balance;
+
+// Enable `dev_mode` for this pallet.
+#[frame_support::pallet(dev_mode)]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::config]
+	pub trait Config: pallet_balances::Config + frame_system::Config {
+		/// The overarching event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+	}
+
+	// Simple declaration of the `Pallet` type. It is placeholder we use to implement traits and
+	// method.
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		// No need to define a `weight` attribute here because of `dev_mode`.
+		pub fn add_dummy(origin: OriginFor<T>, id: T::AccountId) -> DispatchResult {
+			ensure_root(origin)?;
+
+			let dummy = id.clone();
+			Dummy::<T>::mutate(
+				|may_be_dummies| -> DispatchResult {
+					if let Some(dummies) = may_be_dummies.as_mut() {
+						dummies.push(dummy);
+					} else {
+						let mut v = vec![];
+						v.push(dummy);
+						*may_be_dummies = Some(v);
+					}
+					Ok(())
+				},
+			)?;
+
+			// Let's deposit an event to let the outside world know this happened.
+			Self::deposit_event(Event::AddDummy { account: id });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(1)]
+		// No need to define a `weight` attribute here because of `dev_mode`.
+		pub fn set_bar(
+			origin: OriginFor<T>,
+			#[pallet::compact] new_value: T::Balance,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+
+			// Put the new value into storage.
+			<Bar<T>>::insert(&sender, new_value);
+
+			Self::deposit_event(Event::SetBar { account: sender, balance: new_value });
+
+			Ok(())
+		}
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		AddDummy {
+			account: T::AccountId,
+		},
+		SetBar {
+			account: T::AccountId,
+			balance: BalanceOf<T>,
+		},
+	}
+
+	// The MEL requirement for bounded pallets is skipped by `dev_mode`.
```

Comment:
Please explain here and below what the difference is what the default behaviour does.  
(in this case; not generating storage info).